Packs/UBIRCH/Integrations/UBIRCH/UBIRCH.py

```diff
@@ -0,0 +1,340 @@
+        """
+        Callback function when a MQTT client subscribes to a message from the server
+
+        Create incidents, when the client subscribes to an error from the mqtt server.
+        """
+        demisto.info(f"on message. {message.topic} {message.qos} {message.payload}")
+        incidents = create_incidents(message.payload.decode("utf-8"))  # the message payload is binary.
+        demisto.info(f"catch an incident. {incidents}")
+        demisto.createIncidents(incidents)
+
+    try:
+        client.connect(on_connect)
+        client.subscribe(on_message)
+        client.loop_forever()
+    except Exception as e:
+        demisto.error(f'An error occurred in the long running loop: {e}')
+    finally:
+        client.loop_stop()
+
+
+def test_module(client: Client) -> None:
+    """Check if the user configuration is correct
+
+    Return:
+        None: no data returned
+    """
+
+    def on_connect(_client: mqtt.Client, _userdata: dict, _flags: dict, rc: int) -> None:
+        """
+        Callback function when a MQTT client connects to the server
+
+        Check if the connection is succeeded.
+        Stop the loop regardless of whether the connection is succeeded or not.
+        The rc argument is a connection result.
+        """
+        _client.disconnect()
+        _client.loop_stop()
+        if rc != mqtt.MQTT_ERR_SUCCESS:
+            demisto.info(mqtt.connack_string(rc))
+            raise paho.mqtt.MQTTException(mqtt.connack_string(rc))
+        else:
+            demisto.info("connection was succeeded for test")
+
+    try:
+        client.connect(on_connect)
+        client.subscribe()
+        client.loop_forever()
+    except Exception as e:
+        raise DemistoException(
+            f"Test failed. Please check your parameters. \n {e}")
+    demisto.results('ok')
+
+
+def create_sample_incidents() -> None:
+    """Extract sample events stored in the integration context and create them as incidents
+
+    Return:
+        None: no data returned
+    """
+    integration_context = get_integration_context()
+    sample_events = integration_context.get('sample_events')
+    if sample_events:
+        try:
+            incidents = [{'name': "sample_event", 'rawJSON': json.dumps(event)} for event in json.loads(sample_events)]
+            demisto.createIncidents(incidents)
+        except json.decoder.JSONDecodeError as e:
+            raise ValueError(f'Failed deserializing sample events - {e}')
+    else:
+        incidents = [{
+            'name': 'sample incident.',
+        }]
+        demisto.createIncidents(incidents)
+
+
+def main() -> None:
+    """ main function, parses params and runs command functions """
+
+    params = demisto.params()
+    username = params.get('credentials', {}).get('identifier')
+    password = params.get('credentials', {}).get('password')
+    tenant_id = params.get('tenant_id')
+    mqtt_host = params.get('url')
+    mqtt_port = params.get('port')
```

Comment:
You mean changing the credentials part like this?
```
 username = params.get('credentials').get('identifier')
 password = params.get('credentials').get('password')
```

Reply:
That's one way if you'd like to stick with the `.get` methods, but also using `params['url']` can work since it's a mandatory parameter.
Your call here